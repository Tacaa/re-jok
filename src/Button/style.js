import styled, {css, keyframes} from 'styled-components';

import {Transition, hexa, shadeColor} from '../globals'

export const baseButton = css `
    display: flex;
    flex: none;
    align-self: center;
    align-items: center;
    justify-content: center;
    word-break: keep-all;
    cursor: pointer;
    position: relative;
    outline: none;
    border: none;
    text-decoration: none;
    text-align: center;

    font-size: ${props => props.theme.size[props.size].fontSize};

      div + span,
      span + span {
        margin: 0 .8rem;
      }




// =======================Button=====================


`
const classicAnimated = css `
&:hover {
    transform: translateY(-.3rem);
    box-shadow: 0 1rem 2rem rgba(0,0,0,.2);

    &::after {
        transform: scaleX(1.4) scaleY(1.6);
        opacity: 0;
    }
}


&:active {
    outline: none;
    transform: translateY(-.1rem);
    box-shadow: 0 .5rem 1rem rgba(0,0,0,.2);
}
`

const basicAnimated = css `
&:hover{
  border-color:${props => props.backgroundColor
        ? hexa(props.backgroundColor, 0.9)
        : props.primary
          ? hexa(props.theme.primary.main, 0.9)
          : props.secondary
            ? hexa(props.theme.secondary.main, 0.9)
            : "#fff"};
  box-shadow: ${props => props.backgroundColor
              ? (`0 14px 26px -12px ${hexa(props.backgroundColor, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.backgroundColor, 0.2)}`)

              : props.primary
                ? (`0 14px 26px -12px ${hexa(props.theme.primary.main, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.theme.primary.main, 0.2)}`)

                : props.secondary
                  ? (`0 14px 26px -12px ${hexa(props.theme.secondary.main, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.theme.secondary.main, 0.2)}`)

                  : (`0 14px 26px -12px rgba(0, 0, 0, 0.42), 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)`)};

}

&:active{
  box-shadow: -.1rem .1rem .4rem rgba(0, 0, 0, 0.4);
}
`

export const Btn = styled.button.attrs({
  href: props => props.hrefString
    ? props.hrefString
    : null,
  type: props => props.tag === "button"
    ? props.rule
    : null,
  disabled: props => props.disabled || props.loading
})`
  ${baseButton}


  background-color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "#fff"};

  color: ${props => (props.backgroundColor || props.primary || props.secondary)
        ? "#fff"
        : "#000000a6"};

    width: ${props => props.fullWidth && props.shape !== "circle"
          ? "100%"
          : "unset"};


  border: .1rem solid rgba(0,0,0,.15);
  border-color: ${props => props.backgroundColor
            ? props.backgroundColor
            : props.primary
              ? props.theme.primary.main
              : props.secondary
                ? props.theme.secondary.main
                : "#d9d9d9"};


  ${
props => !props.shape
  ? css `
    padding: ${props => props.theme.size[props.size].padding};
    border-radius: ${props => props.theme.size[props.size].borderRadius};
    line-height: ${props => props.theme.size[props.size].lineHeight};

      `
  : props.shape === "circle" && props.variant !== 'icon'
    ? css `
      width: ${props => props.theme.size[props.size].width};
      height: ${props => props.theme.size[props.size].height};
      border-radius: 50%;
      padding: 0;
    `
    : css `
        border-radius: 10rem;
        padding: ${props => props.theme.size[props.size].padding};
        line-height: ${props => props.theme.size[props.size].lineHeight};
    `}

     transition: ${Transition.base.on};
${
  props => props.variant === 'icon'
  ? null
  : props.variant === "outline"
  ? null
  : (
    css`
    &:hover{
      background-color: ${props => props.backgroundColor
      ? hexa(props.backgroundColor, 0.9)
      : props.primary
        ? hexa(props.theme.primary.main, 0.9)
        : props.secondary
          ? hexa(props.theme.secondary.main, 0.9)
          : "#fff"};

    }
    `
    )
}

&::before {
position: absolute;
top: -1px;
left: -1px;
bottom: -1px;
right: -1px;
background: #fff;
opacity: .35;
content: "";
border-radius: inherit;
z-index: 1;
-webkit-transition: opacity .2s;
transition: opacity .2s;
pointer-events: none;
display: ${props => props.disabled || props.loading
      ? "block"
      : 'none'};
}

${props => props.variant === "raised"
        ? (
          !props.animated && props.hasBoxShadow
          ? basicAnimated
          : props.animated === "classic"
            ? classicAnimated
            : null)
        : null}

`

//==================== Outline Button ====================
export const OutlineBtn = Btn.extend `
  background-color: #fff;
  border-color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "rgba(0,0,0,.16)"};
  color: ${props => props.backgroundColor
        ? props.backgroundColor
        : props.primary
          ? props.theme.primary.main
          : props.secondary
            ? props.theme.secondary.main
            : "#0000008a"};

  ${
props => props.variant === "outline"
  ? css `
       &:hover{
            border-color: ${props => props.backgroundColor
    ? shadeColor(props.backgroundColor, -15)
    : props.primary
      ? shadeColor(props.theme.primary.main, -15)
      : props.secondary
        ? shadeColor(props.theme.secondary.main, -15)
        : "rgba(0,0,0,.54)"};

            color: ${props => props.backgroundColor
          ? shadeColor(props.backgroundColor, -15)
          : props.primary
            ? shadeColor(props.theme.primary.main, -15)
            : props.secondary
              ? shadeColor(props.theme.secondary.main, -15)
              : "rgba(0,0,0,.68)"};
       }
     `
  : props.variant === "inverted"
    ? (
      css `
         &:hover{
           background-color: ${props => props.backgroundColor
      ? props.backgroundColor
      : props.primary
        ? props.theme.primary.main
        : props.secondary
          ? props.theme.secondary.main
          : "rgba(0,0,0,.54)"};
            color: #fff;
         }
         `)
    : null}

    ${ props => props.hasBoxShadow ?
      basicAnimated : null
    }



`

//=================== Icon Button =======================
export const IconBtn = Btn.extend `
   border: none;
   color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "#0000008a"};
   width: ${props => props.theme.size[props.size].width};
   height: ${props => props.theme.size[props.size].height};
   border-radius: 50%;
   padding: 0;
   background-color: transparent;
   &:hover{
        border-color: ${props => props.backgroundColor
        ? shadeColor(props.backgroundColor, -15)
        : props.primary
          ? shadeColor(props.theme.primary.main, -15)
          : props.secondary
            ? shadeColor(props.theme.secondary.main, -15)
            : "rgba(0,0,0,.54)"};

        color: ${props => props.backgroundColor
              ? shadeColor(props.backgroundColor, -15)
              : props.primary
                ? shadeColor(props.theme.primary.main, -15)
                : props.secondary
                  ? shadeColor(props.theme.secondary.main, -15)
                  : "rgba(0,0,0,.68)"};
   }

`

//==================== Text Button =======================

const animationTextButton = css`
  &:before{
    content: "";
position: absolute;
bottom: 0;
left: 0;
right: 100%;
border-bottom: 1px solid #222;
transition: right .3s ease-out;
  }
  &:hover:before{
    right: 0;
  }
`


export const TextBtn = styled.button.attrs({
  href: props => props.hrefString
    ? props.hrefString
    : null,
  type: props => props.tag === "button"
    ? props.rule
    : null,
  disabled: props => props.disabled || props.loading
})`
 ${baseButton}
 letter-spacing: 0.08rem;
 background: transparent;
 background-image: none;

 color: ${props => props.color
    ? props.color
    : props.primary
      ? props.theme.primary.main
      : props.secondary
        ? props.theme.secondary.main
        : "#000000a6"};

 padding: ${props => props.theme.size[props.size].padding};
 &:hover{

   color: ${props => props.colorWhenHover ?
        hexa(props.colorWhenHover)
        :
         props.color
          ? hexa(props.color, 0.8)
          : props.primary
            ? hexa(props.theme.primary.main, 0.8)
            : props.secondary
              ? hexa(props.theme.secondary.main, 0.8)
              : hexa("#000000a6", 0.8)};

    border-color: transparent;

 }
 &:active{
   color: ${props => props.colorWhenHover ?
        hexa(props.colorWhenHover)
        : props.color
                  ? hexa(props.color, 0.8)
                  : props.primary
                    ? hexa(props.theme.primary.main, 0.8)
                    : props.secondary
                      ? hexa(props.theme.secondary.main, 0.8)
                      : hexa("#000000a6", 0.8)};
border-color: transparent;
 }

${
  props =>
   props.hoverable && animationTextButton
}



`


// =================== Sub Component for Button ========================

export const Label = styled.span `
  display: block;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  align-self: center;
  margin: auto;
`;

const spin = keyframes `
  to {transform: rotate(360deg);}
`;

export const SpinnerButton = styled.div `
display: inline-block;
width: ${props => props.theme.size[props.size].fontSize};
height: ${props => props.theme.size[props.size].fontSize};
min-width: ${props => props.theme.size[props.size].fontSize};
min-height: ${props => props.theme.size[props.size].fontSize};
position: ${props => props.shape === "circle"
  ? ""
  : "relative"};
color: inherit;
flex: 0 0 ${props => props.theme.size[props.size].fontSize};
order: ${props => props.shape !== "circle"
    ? props.iconPosition === "left"
      ? 0
      : 1
    : 0};
  &::before{
    content: '';
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  left: 50%;
  width: ${props => props.theme.size[props.size].fontSize};
  height: ${props => props.theme.size[props.size].fontSize};
  margin-top: calc(${props => props.theme.size[props.size].fontSize} / -2);
  margin-left: calc(${props => props.theme.size[props.size].fontSize} / -2);
  border-radius: 50%;
  border-color: ${props => props.primary || props.secondary
      ? "#fff"
      : "#d9d9d9"};
  border-top: 2px solid;
  border-right: 2px solid transparent;
  animation: ${spin} .6s linear infinite;
  }
`
