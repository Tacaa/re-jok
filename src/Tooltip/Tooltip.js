// @flow
import * as React from 'react';

import {
  StyledTooltip
} from './style';

type Props = {
  children?: any,
  position: 'top' | 'left' | 'right' | 'bottom',
  label?: string
}

const defaultProps = {
  position: 'bottom'
}

class Tooltip extends React.Component<Props>{

  static defaultProps = defaultProps;

  render(){
    const {
      children,
      ...rest
    } = this.props;
    return (
      <StyledTooltip {...rest}>
        {children}
      </StyledTooltip>
    )
  }
}

export default Tooltip
