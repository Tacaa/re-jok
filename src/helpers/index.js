export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
