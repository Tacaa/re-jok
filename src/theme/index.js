// @flow
// export our theme to the provider
// This is globally available in styled-components when interpolating a function like so:
// ${(props) => props.theme}
// Or using import { withTheme } from 'styled-components';


export const defaultTheme = {
  size: {
    small: {
      fontSize: '1.2rem',
      padding: '0.5rem 0.9rem',
      "lineHeight": "1",
      "borderRadius": ".2rem",
      width: "2.4rem",
      height: "2.4rem"
    },
    default: {
      fontSize: '1.4rem',
      padding: '0.9rem 1.8rem',
      lineHeight: "1.5",
      borderRadius: ".4rem",
      width: "3.2rem",
      height: "3.2rem"
    },
    medium: {
      fontSize: '1.8rem',
      padding: '1.4rem 2.7rem',
      lineHeight: "1.5",
      borderRadius: ".6rem",
      width: "4rem",
      height: "4rem"

    },
    large: {
      fontSize: '2rem',
      padding: '1.8rem 3.6rem',
      lineHeight: "2",
      borderRadius: ".6rem",
      width: "4.8rem",
      height: "4.8rem"
    }
  },
  grid:{
    xs: 768,
    sm: 768,
    md: 992,
    lg: 1200,
    xl: 1920
  },
  text: {
    primary: "rgba(0, 0, 0, 0.85)",
    secondary: "rgba(0, 0, 0, 0.48)",
    hint: "rgba(0, 0, 0, 0.38)"
  },
  background: {
    paper: "#fff",
    default: "#fafafa"
  },
  typography: {
    title: {
      fontSize: "2rem",
      fontWeight: "600",
      lineHeight: "2.8rem"

    },
    subtitle: {
      fontSize: "1.6rem",
      fontWeight: "400",
      lineHeight: "2.4rem"
    }
  },
  primary: {
    main: "#1c8ffb",

  },
  secondary: {
    main: "#1b1c1d",
  },
  info: {
    main: "#1890ff"
  },
  error: {
    main: "#f5222d"
  },
  success: {
    main: "#52c41a"
  },
  warning: {
    main: "#faad14"
  },
  zIndex: {
    base: 1,
    background: 0,
    hidden: -1,
    card: 2,
    loading: 3,
    avatar: 3,
    form: 3,
    search: 3,
    badge: 100,
    modal: 1500,
  }
};

const changeTheme = (defaultTheme, theme)=>{
  let cloneTheme = {...defaultTheme};


  Object.keys(theme).forEach(function (key) {
    if ( 'object' === typeof theme[key] && !Array.isArray(theme[key]) ) {
    	changeTheme(cloneTheme[key], theme[key])
    }
    else if ( 'string' === typeof theme[key] || 'number' === typeof theme[key] ) {
      cloneTheme[key] = theme[key]
    }
  })
  return cloneTheme;
}

export const setTheme = (theme: Object) => {

   const newTheme = changeTheme(defaultTheme, theme);

   return newTheme;

}
