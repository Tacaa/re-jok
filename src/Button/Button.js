// @flow
import * as React from 'react'

import Icon from '../Icon';

import {
  Btn,
  OutlineBtn,
  Label,
  SpinnerButton,
  IconBtn
} from './style';

import { isChild } from '../helpers/typeUtils';

type Props = {
  /** Set button unable to click or not **/
  disabled?: boolean,
  /** Event when user clicked to button **/
  onClick?: Function,
  /** Set button to a link**/
  hrefString?: string,
  /** The size of button . The default size of button is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of button **/
  shape?: 'circle' | 'rounded',
  /** Button can show a loading**/
  loading?: boolean,
  /** Set icon in button**/
  icon?: string,
  /** Add more style to button **/
  style?: Object,
  /** Add more class to button **/
  className?: string,
  /** Primary color of button **/
  primary?: boolean,
  /** Secondary color of button **/
  secondary?: boolean,
  /** Set width of button to 100%**/
  fullWidth?: boolean,
  /** move icon to left or right label. The default is left **/
  iconPosition: 'left' | 'right',
  /** if you don't want use primary or secondary , you can set your color to button **/
  backgroundColor?: string,
  /** Make your button animated**/
  animated?: string,
  /** Change tag name of button ... span, div, a v..v **/
  tag?: string,
  /** Change type of button**/
  variant: 'outline' | 'inverted' | 'raised' | 'icon',
  /** create button with box shadow or not**/
  hasBoxShadow: boolean,
  /** Set type of button **/
  rule: 'button' | 'reset' | 'submit',
  /**Children element can be string, number, HTML node anything you want**/
  children?: any,
}

const defaultProps = {
  disabled: false,
  size: 'default',
  rule: 'button',
  loading: false,
  primary: false,
  secondary: false,
  fullWidth: false,
  iconPosition: 'left',
  backgroundColor: "",
  tag: "button",
  variant: 'raised',
  hrefString: "",
  hasBoxShadow: false
};

class Button extends React.Component<Props> {
  static defaultProps = defaultProps;

  render() {
    const {
      tag,
      hrefString,
      children,
      loading,
      shape,
      size,
      primary,
      secondary,
      icon,
      iconPosition,
      variant,

    } = this.props;

   const ListButton = {
     raised: Btn,
     outline: OutlineBtn,
     inverted: OutlineBtn,
     icon: IconBtn
   }
   const hasChild = !isChild(children);

    const ElementType = !hrefString ? ListButton[variant].withComponent(tag) : ListButton[variant].withComponent('a')

    return (
      <ElementType {...this.props}>
        {
          icon
            ? (
              loading
              ? (<SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>)
              : (<Icon iconPosition={iconPosition} name={icon}/>))
            : ('')
        }
        {loading && !icon && <SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>}
        {
          shape === "circle" || variant === "icon"
            ? (null)
            : (hasChild && <Label>
              {children}
            </Label>)
        }

      </ElementType>
    )
  }
}

export default Button
