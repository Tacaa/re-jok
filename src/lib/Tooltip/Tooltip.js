// @flow
import * as React from 'react';

import {
  StyledTooltip
} from './style';

import { isChild } from '../helpers/typeUtils';

type Props = {
  /** Children of tooltip could be anything**/
  children?: any,
  /** Set position of tooltip box.. default is bottom**/
  position: 'top' | 'left' | 'right' | 'bottom' | 'topRight' | 'topLeft' | 'bottomRight' | 'bottomLeft',
  /** Label of tooltip**/
  label?: any,
  /** Set color of tooltip box**/
  tooltipColor: string,
}

const defaultProps = {
  position: 'bottom',
  tooltipColor: 'rgba(0, 0, 0, .75)'
}

class Tooltip extends React.Component<Props>{

  static defaultProps = defaultProps;

  render(){
    const {
      children,
      ...rest
    } = this.props;

    const hasChild = !isChild(children);

    return (
      <StyledTooltip {...rest}>
        {hasChild && children}
      </StyledTooltip>
    )
  }
}

export default Tooltip
