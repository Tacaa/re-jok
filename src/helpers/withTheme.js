import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
