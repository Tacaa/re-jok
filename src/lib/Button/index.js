import Button from './Button';

import TextButton from './TextButton';


export default {
  Button,
  TextButton
};