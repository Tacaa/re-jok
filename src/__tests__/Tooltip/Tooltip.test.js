import React from 'react';
import Tooltip from '../../Tooltip/Tooltip';
import Button from '../../Button/Button';

import {defaultTheme} from '../../theme';

import {shallowWithTheme, mountWithTheme, renderWithTheme} from '../../helpers/withTheme';
import 'jest-enzyme';
import 'jest-styled-components';

describe('<Tooltip />', () => {
  it('should render', () => {
    const tree = renderWithTheme(
      <Tooltip label="Tooltip">
        <Button>Button</Button>
      </Tooltip>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with position', () => {
    const tree = renderWithTheme(
      <Tooltip label="Tooltip position" position="top">
        <Button>Button</Button>
      </Tooltip>
    ).toJSON();
    expect(tree).toMatchSnapshot();


  });


  it('should render with position and color', () => {
    const tree = renderWithTheme(
      <Tooltip label="Tooltip position" tooltipColor="#11D3BC" position="top">
        <Button>Button</Button>
      </Tooltip>
    ).toJSON();
    expect(tree).toMatchSnapshot();


  });



})
