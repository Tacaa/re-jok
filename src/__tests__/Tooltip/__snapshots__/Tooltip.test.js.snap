// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render 1`] = `
.c0 {
  position: relative;
  font-size: 1.4rem;
  line-height: 1.5;
  color: rgba(0,0,0,.64);
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  display: block;
  visibility: visible;
}

.c0::after {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  max-width: 25rem;
  min-width: 4rem;
  padding: .6rem .8rem;
  color: #fff;
  text-align: left;
  z-index: 3000;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,0,0,.75);
  border-radius: .4rem;
  box-shadow: 0 .2rem .8rem rgba(0,0,0,.15);
  min-height: 3.2rem;
  content: 'Tooltip';
}

.c0::before {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  content: "";
  border-width: .5rem;
  border-image: initial;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  z-index: 3000;
}

.c0::after {
  top: 100%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  left: 50%;
}

.c0::before {
  bottom: 100%;
  top: calc(100% - 5px);
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  border-top-width: 0px;
  border-bottom-color: rgba(0,0,0,.75);
}

.c0:hover:after,
.c0:hover:before {
  opacity: 1;
  -webkit-transform: translateX(-50%) translateY(5px);
  -ms-transform: translateX(-50%) translateY(5px);
  transform: translateX(-50%) translateY(5px);
}

.c2 {
  display: block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  word-break: keep-all;
  cursor: pointer;
  position: relative;
  outline: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  font-size: 1.4rem;
  background-color: #fff;
  color: rgba(0,0,0,.68);
  width: unset;
  border: .1rem solid transparent;
  border-color: #d9d9d9;
  padding: 0.9rem 1.8rem;
  border-radius: .4rem;
  line-height: 1.5;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c1 div + span,
.c1 span + span {
  margin: 0 .8rem;
}

.c1:hover {
  background-color: #fff;
}

.c1::before {
  position: absolute;
  top: -1px;
  left: -1px;
  bottom: -1px;
  right: -1px;
  background: #fff;
  opacity: .35;
  content: "";
  border-radius: inherit;
  z-index: 1;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  pointer-events: none;
  display: none;
}

<div
  className="c0"
  label="Tooltip"
>
  <button
    className="c1"
    disabled={false}
    href={null}
    size="default"
    type="button"
  >
    
    <span
      className="c2"
    >
      Button
    </span>
  </button>
</div>
`;

exports[`<Tooltip /> should render with position 1`] = `
.c0 {
  position: relative;
  font-size: 1.4rem;
  line-height: 1.5;
  color: rgba(0,0,0,.64);
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  display: block;
  visibility: visible;
}

.c0::after {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  max-width: 25rem;
  min-width: 4rem;
  padding: .6rem .8rem;
  color: #fff;
  text-align: left;
  z-index: 3000;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,0,0,.75);
  border-radius: .4rem;
  box-shadow: 0 .2rem .8rem rgba(0,0,0,.15);
  min-height: 3.2rem;
  content: 'Tooltip position';
}

.c0::before {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  content: "";
  border-width: .5rem;
  border-image: initial;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  z-index: 3000;
}

.c0::after {
  bottom: 100%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  left: 50%;
}

.c0::before {
  bottom: calc(100% - 5px);
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  left: 50%;
  border-bottom-width: 0px;
  border-top-color: rgba(0,0,0,.75);
}

.c0:hover:after,
.c0:hover:before {
  opacity: 1;
  -webkit-transform: translateX(-50%) translateY(-5px);
  -ms-transform: translateX(-50%) translateY(-5px);
  transform: translateX(-50%) translateY(-5px);
}

.c2 {
  display: block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  word-break: keep-all;
  cursor: pointer;
  position: relative;
  outline: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  font-size: 1.4rem;
  background-color: #fff;
  color: rgba(0,0,0,.68);
  width: unset;
  border: .1rem solid transparent;
  border-color: #d9d9d9;
  padding: 0.9rem 1.8rem;
  border-radius: .4rem;
  line-height: 1.5;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c1 div + span,
.c1 span + span {
  margin: 0 .8rem;
}

.c1:hover {
  background-color: #fff;
}

.c1::before {
  position: absolute;
  top: -1px;
  left: -1px;
  bottom: -1px;
  right: -1px;
  background: #fff;
  opacity: .35;
  content: "";
  border-radius: inherit;
  z-index: 1;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  pointer-events: none;
  display: none;
}

<div
  className="c0"
  label="Tooltip position"
>
  <button
    className="c1"
    disabled={false}
    href={null}
    size="default"
    type="button"
  >
    
    <span
      className="c2"
    >
      Button
    </span>
  </button>
</div>
`;

exports[`<Tooltip /> should render with position and color 1`] = `
.c0 {
  position: relative;
  font-size: 1.4rem;
  line-height: 1.5;
  color: rgba(0,0,0,.64);
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  display: block;
  visibility: visible;
}

.c0::after {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  max-width: 25rem;
  min-width: 4rem;
  padding: .6rem .8rem;
  color: #fff;
  text-align: left;
  z-index: 3000;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #11D3BC;
  border-radius: .4rem;
  box-shadow: 0 .2rem .8rem rgba(0,0,0,.15);
  min-height: 3.2rem;
  content: 'Tooltip position';
}

.c0::before {
  position: absolute;
  opacity: 0;
  text-transform: none;
  -webkit-transition: all 0.216s ease;
  transition: all 0.216s ease;
  content: "";
  border-width: .5rem;
  border-image: initial;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  z-index: 3000;
}

.c0::after {
  bottom: 100%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  left: 50%;
}

.c0::before {
  bottom: calc(100% - 5px);
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  left: 50%;
  border-bottom-width: 0px;
  border-top-color: #11D3BC;
}

.c0:hover:after,
.c0:hover:before {
  opacity: 1;
  -webkit-transform: translateX(-50%) translateY(-5px);
  -ms-transform: translateX(-50%) translateY(-5px);
  transform: translateX(-50%) translateY(-5px);
}

.c2 {
  display: block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  word-break: keep-all;
  cursor: pointer;
  position: relative;
  outline: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  font-size: 1.4rem;
  background-color: #fff;
  color: rgba(0,0,0,.68);
  width: unset;
  border: .1rem solid transparent;
  border-color: #d9d9d9;
  padding: 0.9rem 1.8rem;
  border-radius: .4rem;
  line-height: 1.5;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c1 div + span,
.c1 span + span {
  margin: 0 .8rem;
}

.c1:hover {
  background-color: #fff;
}

.c1::before {
  position: absolute;
  top: -1px;
  left: -1px;
  bottom: -1px;
  right: -1px;
  background: #fff;
  opacity: .35;
  content: "";
  border-radius: inherit;
  z-index: 1;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  pointer-events: none;
  display: none;
}

<div
  className="c0"
  label="Tooltip position"
>
  <button
    className="c1"
    disabled={false}
    href={null}
    size="default"
    type="button"
  >
    
    <span
      className="c2"
    >
      Button
    </span>
  </button>
</div>
`;
